cloudbridge/cloud/factory.py

```diff
@@ -27,9 +79,14 @@ class CloudProviderFactory(object):
     """
 
     def __init__(self):
+        self._middleware = ParentMiddlewareManager()
```

Comment:
I think we need a different name because ParentMiddlewareManager doesn't seem to explain what it's doing. Also, there is the issue of namespacing. And finally, I guess there's the issue of being able to intercept behaviour of a particular cloud (say openstack), and change behaviour for all classes of that cloud. One way would be to prefix the cloudname to the event. Another way would be is to check the instancetype of the sender: `if isinstance(OpenStackProvider, event_args['sender'].provider)` or some such thing. I think namespacing would be good as long as isolation is maintained.